src/client/Client.cc

```diff
@@ -10960,7 +11011,8 @@ int Client::lazyio_synchronize(int fd, loff_t offset, size_t count)
 // =============================
 // snaps
 
-int Client::mksnap(const char *relpath, const char *name, const UserPerm& perm)
+int Client::mksnap(const char *relpath, const char *name, const UserPerm& perm,
+                   mode_t mode, const char *metadata_dict)
```

Comment:
ditto

Reply:
How would python bindings pass in metadata if this accepts a std::map?

Reply:
Just as there's Cython to aid with Python-to-C marshaling, there are quite a few Python-to-C++ glue libraries available as well eg. boost::Python

I think we could use one of those in this case.

Reply:
maybe. boost::Python is probably more suited for complex uses -- the case here is pretty straightforward. In fact, librados C C API uses this format (key1\0value1\0key2\0value2\0\0) as metadata to be accepted.

Reply:
The python bindings use libcephfs. The C-friendly interface glue should go in libcephfs, not in Client.